src/media-segment-request.js

```diff
@@ -516,7 +519,19 @@ const handleSegmentBytes = ({
             message.logs.forEach(function(log) {
               onTransmuxerLog(merge(log, {stream: 'mp4CaptionParser'}));
             });
-            finishLoading(message.captions);
+
+            workerCallback({
+              action: 'probeEmsgID3',
+              data: bytesAsUint8Array,
+              transmuxer: segment.transmuxer,
+              offset: startTime,
+              callback: ({emsgData, id3Frames}) => {
+                // transfer bytes back to us
+                bytes = emsgData.buffer;
+                segment.bytes = bytesAsUint8Array = emsgData;
+                finishLoading(message.captions, id3Frames);
+              }
+            });
```

Comment:
Curious: Currently, we have three `workerCallbacks` here chaining each other. It seems like they can be run absolutely separately. Should we separate them?

Reply:
I couldn't find a way to separate them completely without breaking the `startTime` and `captions` probe for some reason. After playing around with it a bit I settled on nesting the id3 probing similar to the way the captions extraction is nested. I'll play around with it a bit more and see if there's a better way to pull them all apart, I think that we should separate them if possible.

Reply:
I tried a few other permutations of splitting these `workerCallback`s and am still running into issues sending messages to the service worker. Going to leave it as is for now, and make a note and possibly an issue to look into splitting up those calls backs.